Nautilus/Handlers/PrefabHandler.cs

```diff
@@ -69,7 +76,18 @@ private static IEnumerator ProcessPrefabAsync(GameObject obj, PrefabInfo info, P
         if (Prefabs.TryGetPostProcessorForInfo(info, out var postProcessor))
             yield return postProcessor?.Invoke(obj);
 
-        ModPrefabCache.AddPrefab(obj);
+
+        if (ModPrefabCache.TryGetPrefabFromCache(info.ClassID, out var obj2))
+        {
+            gameObject.Set(obj2);
+            GameObject.Destroy(obj);
```

Comment:
This looks dangerous. Since non-instantiated asset bundle prefabs can be custom prefabs, wouldn't destroying a prefab would make the current one also null?

Reply:
> This seems fine for now, should be ready to merge, but it would probably be best to make a system that uses existing prefab requests for “work in progress” prefabs rather than running multiple requests at the same time.

> This looks dangerous. Since non-instantiated asset bundle prefabs can be custom prefabs, wouldn't destroying a prefab would make the current one also null?

Ohh ooohhhhhhhh   brain explosion..... 

Forgot that addressables Cache their request calls so as not to duplicate them.

I need to add a modPrefabRequest cache and it will fix both of these issues.

Thanx guys.